File: resources/scop_1.5.1/java/examples/XMLReceiver.java
// XMLReciever.java - DMI - 25-12-02

/* Usage: java XMLReciever [-inspect] */

import java.io.*;

public class XMLReceiver
{
	public static void main(String[] args)
	{
		SCOPXML scop;
		AddressBook ab;
		Vertex v;

		scop = new SCOPXML("localhost", "XMLReceiver");
		v = scop.get_struct();
		if(args.length == 1 && args[0].equals("-inspect"))
		{
			String s = Vertex.pretty_print(v);
			System.out.println(s);
		}
		ab = new AddressBook(v);
		ab.dump();
		scop.close();
	}
}

File: resources/scop_1.5.1/examples/xml_receiver.cpp
// xml_reciever.cpp - DMI - 7-9-02

/* Usage: xml_reciever [-inspect] */

#include <scop.h>
#include <scopxml.h>

#include "address_book.h"

int main(int argc, char **argv)
{
	int sock;
	AddressBook *ab;
	vertex *v;
	
	sock = scop_open("localhost", "xml_receiver");
	v = scop_get_struct(sock);
	if(argc == 2 && !strcmp(argv[1], "-inspect"))
	{
		char *c = pretty_print(v);
		printf("%s\n", c);
		delete[] c;
	}
	ab = new AddressBook(v);
	delete v;
	ab->dump();
	delete ab;
	
	close(sock);
	return 0;
}
